### PR TITLE
Make DSC's component conditions user friendly

### PR DESCRIPTION
**Pull request by lburgazzoli**
## Description

- **Make DSC's component conditions user friendly**

```yaml
  conditions:
  - lastHeartbeatTime: "2024-12-13T13:34:16Z"
    lastTransitionTime: "2024-12-13T13:33:58Z"
    message: Not Available
    reason: Unknown
    status: "False"
    type: CodeFlareReady
  - lastHeartbeatTime: "2024-12-13T13:34:16Z"
    lastTransitionTime: "2024-12-13T13:33:58Z"
    message: Component ManagementState is set to Removed
    reason: Removed
    status: "False"
    type: DashboardReady
  - lastHeartbeatTime: "2024-12-13T13:34:17Z"
    lastTransitionTime: "2024-12-13T13:33:59Z"
    message: Component ManagementState is set to Removed
    reason: Removed
    status: "False"
    type: KserveReady
  - lastHeartbeatTime: "2024-12-13T13:34:17Z"
    lastTransitionTime: "2024-12-13T13:33:59Z"
    message: Component ManagementState is set to Removed
    reason: Removed
    status: "False"
    type: KueueReady
  - lastHeartbeatTime: "2024-12-13T13:34:17Z"
    lastTransitionTime: "2024-12-13T13:33:59Z"
    message: Component ManagementState is set to Removed
    reason: Removed
    status: "False"
    type: ModelControllerReady
  - lastHeartbeatTime: "2024-12-13T13:34:17Z"
    lastTransitionTime: "2024-12-13T13:33:59Z"
    message: Component ManagementState is set to Removed
    reason: Removed
    status: "False"
    type: ModelMeshServingReady
  - lastHeartbeatTime: "2024-12-13T13:34:17Z"
    lastTransitionTime: "2024-12-13T13:33:59Z"
    message: Component ManagementState is set to Removed
    reason: Removed
    status: "False"
    type: ModelRegistryReady
  - lastHeartbeatTime: "2024-12-13T13:34:17Z"
    lastTransitionTime: "2024-12-13T13:33:59Z"
    message: Component ManagementState is set to Removed
    reason: Removed
    status: "False"
    type: RayReady
  - lastHeartbeatTime: "2024-12-13T13:34:17Z"
    lastTransitionTime: "2024-12-13T13:33:59Z"
    message: DataScienceCluster resource reconciled successfully
    reason: ReconcileCompleted
    status: "True"
    type: ReconcileComplete
  - lastHeartbeatTime: "2024-12-13T13:34:17Z"
    lastTransitionTime: "2024-12-13T13:33:59Z"
    message: Component ManagementState is set to Removed
    reason: Removed
    status: "False"
    type: TrainingOperatorReady
  - lastHeartbeatTime: "2024-12-13T13:34:17Z"
    lastTransitionTime: "2024-12-13T13:33:59Z"
    message: Component ManagementState is set to Removed
    reason: Removed
    status: "False"
    type: TrustyAIReady
  - lastHeartbeatTime: "2024-12-13T13:34:17Z"
    lastTransitionTime: "2024-12-13T13:33:59Z"
    message: Component ManagementState is set to Removed
    reason: Removed
    status: "False"
    type: WorkbenchesReady
```

Requires:
- https://github.com/opendatahub-io/opendatahub-operator/pull/1449
- https://github.com/opendatahub-io/opendatahub-operator/pull/1450

## How Has This Been Tested?

e2e tests have been amended

## Merge criteria

- [X] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [X] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [X] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] ~~Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).~~
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pkg/cluster/gvk/gvk.go [hunk 1]
```diff
@@ -105,39 +107,39 @@ var (
...
 	Kserve = schema.GroupVersionKind{
-		Group:   "components.platform.opendatahub.io",
-		Version: "v1alpha1",
-		Kind:    "Kserve",
+		Group:   componentApi.GroupVersion.Group,
+		Version: componentApi.GroupVersion.Version,
+		Kind:    componentApi.KserveKind,
 	}
 
 	Kueue = schema.GroupVersionKind{
```
Comment (grdryn) [inline pkg/cluster/gvk/gvk.go:145]: Should this one be changed like the others also?

Comment (lburgazzoli) [reply pkg/cluster/gvk/gvk.go:145]: oh yes

Comment (lburgazzoli) [reply pkg/cluster/gvk/gvk.go:145]: done

**Pull request merged**